grype/db/v6/fillers.go

```diff
@@ -0,0 +1,160 @@
+package v6
+
+import (
+	"encoding/json"
+	"errors"
+
+	"gorm.io/gorm"
+)
+
+func affectedCPEVulnerabilityHandles(affectedPackages []*AffectedCPEHandle) []*VulnerabilityHandle {
+	var out []*VulnerabilityHandle
+	for _, h := range affectedPackages {
+		if h == nil {
+			continue
+		}
+		out = append(out, h.Vulnerability)
+	}
+	return out
+}
+
+func affectedPackageVulnerabilityHandles(affectedPackages []*AffectedPackageHandle) []*VulnerabilityHandle {
+	var out []*VulnerabilityHandle
+	for _, h := range affectedPackages {
+		if h == nil {
+			continue
+		}
+		out = append(out, h.Vulnerability)
+	}
+	return out
+}
+
+func fillAffectedCPEHandles(db *gorm.DB, handles []*AffectedCPEHandle) error {
+	return errors.Join(
+		fillRefs(db, handles, affectedCPEVulnerabilityHandleRef, vulnerabilityHandleID),
+		fillRefs(db, handles, affectedCPECPERef, cpeHandleID),
+		fillAffectedCPEBlobs(db, handles),
+	)
+}
+
+func fillAffectedCPEBlobs(db *gorm.DB, handles []*AffectedCPEHandle) error {
+	return fillBlobs(db, handles, affectedCPEBlobRef)
+}
+
+func fillAffectedPackageBlobs(db *gorm.DB, handles []*AffectedPackageHandle) error {
+	return fillBlobs(db, handles, affectedPackageBlobRef)
+}
+
+func fillAffectedPackageHandles(db *gorm.DB, handles []*AffectedPackageHandle) error {
+	return errors.Join(
+		fillAffectedPackageBlobs(db, handles),
+		fillRefs(db, handles, affectedPackageVulnerabilityHandleRef, vulnerabilityHandleID),
+		fillRefs(db, handles, affectedPackageOperatingSystemHandleRef, operatingSystemID),
+		fillRefs(db, handles, affectedPackagePackageHandleRef, packageID),
+	)
+}
+
+func (s vulnerabilityProvider) fillVulnerabilityHandles(handles []*VulnerabilityHandle) error {
+	return errors.Join(
+		s.fillProviders(handles),
+		fillBlobs(s.db, handles, vulnerabilityHandleBlobRef),
+	)
+}
+
+func fillBlobs[ContainingType, BlobType any](db *gorm.DB, handles []*ContainingType, blobRef refProvider[ContainingType, BlobType]) error {
+	for i := range handles {
+		ref := blobRef(handles[i])
+		// if no ID or if the blob is already set, do nothing
+		if ref.id == nil || *ref.ref != nil {
+			continue
+		}
+		var blob Blob
+		err := db.First(&blob, *ref.id).Error
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			continue
+		}
+		if err != nil {
+			return err
+		}
+
+		var t BlobType
+		err = json.Unmarshal([]byte(blob.Value), &t)
+		if err != nil {
+			return err
+		}
+		*ref.ref = &t
+	}
+	return nil
+}
```

Comment:
The blobstore already has `attachBlobValue` which does this functionality

Reply:
hmm. I've probably missed something, but this is not exported -- how am I supposed to do this while narrowing the initial results?